fili-core/src/main/java/com/yahoo/bard/webservice/web/endpoints/EndpointServlet.java

```diff
@@ -2,26 +2,53 @@
 // Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
 package com.yahoo.bard.webservice.web.endpoints;
 
+import static com.yahoo.bard.webservice.web.DefaultResponseFormatType.CSV;
+
 import com.yahoo.bard.webservice.application.ObjectMappersSuite;
-import com.yahoo.bard.webservice.web.apirequest.ApiRequest;
+import com.yahoo.bard.webservice.util.AllPagesPagination;
+import com.yahoo.bard.webservice.util.Pagination;
 import com.yahoo.bard.webservice.web.CsvResponse;
 import com.yahoo.bard.webservice.web.JsonResponse;
-import com.yahoo.bard.webservice.web.util.ResponseFormat;
+import com.yahoo.bard.webservice.web.apirequest.ApiRequest;
+import com.yahoo.bard.webservice.web.apirequest.PageLinkBuilder;
+import com.yahoo.bard.webservice.web.util.ResponseUtils;
 
+import java.util.Collection;
 import java.util.List;
+import java.util.function.Supplier;
 import java.util.stream.Stream;
 
 import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
 import javax.ws.rs.core.HttpHeaders;
 import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.StreamingOutput;
+import javax.ws.rs.core.UriInfo;
 
 /**
  * Abstract class making available the common code between the servlets that serve the endpoints.
  */
 public abstract class EndpointServlet {
+
     protected final ObjectMappersSuite objectMappers;
+    private final ResponseUtils responseUtils;
+    private final Supplier<Response.ResponseBuilder> responseBuilderSupplier;
+
+
+    /**
+     * Constructor.
+     *
+     * @param objectMappers  Shared JSON tools
+     */
+    public EndpointServlet(
+            ObjectMappersSuite objectMappers,
+            Supplier<Response.ResponseBuilder> responseBuilderSupplier
+    ) {
+        this.objectMappers = objectMappers;
+        this.responseUtils = new ResponseUtils();
```

Comment:
inject?

Reply:
I'd like to postpone doing this just because I don't know if that single method utility class is long for this world.